src/D2L.Security.OAuth2/Keys/Default/MonoRsaSecurityKey.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace D2L.Security.OAuth2.Keys.Default {
+
+	internal sealed class MonoRsaSecurityKey : AsymmetricSecurityKey {
+
+		private RSA m_rsa;
+
+		public MonoRsaSecurityKey( RSA rsa ) {
+			m_rsa = rsa;
+		}
+
+		public override byte[] DecryptKey( string algorithm, byte[] keyData ) {
+			throw new NotImplementedException();
+		}
+
+		public override byte[] EncryptKey( string algorithm, byte[] keyData ) {
+			throw new NotImplementedException();
+		}
+
+		public override bool IsAsymmetricAlgorithm( string algorithm ) {
+			switch (algorithm) {
+			case SecurityAlgorithms.DsaSha1Signature:
+			case SecurityAlgorithms.RsaSha1Signature:
+			case SecurityAlgorithms.RsaSha256Signature:
+			case SecurityAlgorithms.RsaOaepKeyWrap:
+			case SecurityAlgorithms.RsaV15KeyWrap:
+				return true;
+			default:
+				return false;
+			}
+		}
+
+		public override bool IsSupportedAlgorithm( string algorithm ) {

```

Comment:
could you also implement this? We did it for ecdsa, https://github.com/Brightspace/D2L.Security.OAuth2/blob/master/src/D2L.Security.OAuth2/Keys/Default/EcDsaSecurityKey.cs#L98